src/java/com/unifina/signalpath/blockchain/Web3jHelper.java

```diff
@@ -201,7 +201,18 @@ public static long getBlockTime(Web3j web3j, TransactionReceipt tr) throws IOExc
 			log.error("Error fetching block " + dbp + ". Error = " + eb.getError());
 			return -1;
 		}
-		long ts = eb.getBlock().getTimestamp().longValue();
+		EthBlock.Block block = eb.getBlock();
+		if(block == null){
+			log.error("getBlock() returned null block for txHash " + tr.getTransactionHash());
+			return -1;
```

Comment:
Returning an unexpected value to represent an error condition probably isn't the best approach. A careless user of `getBlockTime` could easily overlook to check for a value `-1` and accidentally use that as the timestamp. Better would be to throw a declared exception (e.g. `BlockWasNullException`) to force the calling code to handle the situation appropriately.